src/mapgen.cpp

```diff
@@ -10620,6 +10657,64 @@ int map::place_items(items_location loc, int chance, int x1, int y1,
     return item_num;
 }
 
+int map::place_loot( std::string id, int chance, int x1, int y1, int x2, int y2, int ammo, int magazine, int turn )
```

Comment:
It's a bit sad that one needs to pass the id of different things (item groups vs item type) in the same parameter. It defeats the whole purpose of having different types.